static/js/compose_validate.js

```diff
@@ -201,6 +201,12 @@ export function warn_if_topic_resolved(topic_changed) {
             return;
         }
 
+        if (compose_state.has_recipient_viewed_topic_resolved_banner()) {
+            // If the user has already seen the warning once, we don't
+            // show it to him again. We return early in this case
```

Comment:
I think it would be good to add a bit more detail to this comment about the case where this happens -- if the user sees the warning and closes it, we don't want it to reappear.